### ISSUE TITLE
Including ActionView::Helpers in a serializer results in `stack level too deep`.

### ISSUE THREAD

**Issue by 0x962** [jsonapi-serializer/jsonapi-serializer#152]
## Expected Behavior
For a serializer to work as usual even when ActionView::Helpers are included in the class. 

## Actual Behavior
Including ActionView::Helpers results in a `stack level too deep` error. This error is not thrown when using a serializer outside of a controller or a request-response cycle. Running `YourSerializer.new(@your_collection, {})` in a rails console for instance will render no errors.

## Steps to Reproduce the Problem

  1. Create a serializer
  2. Include `ActionView::Helpers`
  3. Stack level too deep

## Specifications

  - Version: 2.0.0
  - Ruby version: ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-linux]

## Example 

```
include ActionView::Helpers
class ProductSerializer < BaseSerializer

   belongs_to :brand, if: Proc.new { |record| record.brand.present? }
   has_one :sku, polymorphic: true

   attributes :description, :share_url
end
```

**Comment by stas** [comment]
@0x962 I'm not sure I'm able to replicate the issue. As you can see below, my serializer uses `ActionView::Helpers` without any issues. Could you provide a backtrace or more helpful information please?

```ruby
(byebug) ActivitySerializer.new(activity).to_hash
{:data=>{:id=>"ab530f49-47df-4208-af21-0238031bb66e", :type=>:activity, :attributes=>{:action=>"POST", :client=>"browser", :os=>"Windows with SQL Server Web (Amazon VPC)", :misc=>{}, :created_at=>Sun, 06 Dec 2020 18:24:34 UTC +00:00, :remote_address=>"144.91.96.43"}, :relationships=>{:user=>{:data=>{:id=>"f953401d-37d4-4780-b471-5c8d4596ab24", :type=>:customer}}, :resource=>{:data=>{:id=>"f953401d-37d4-4780-b471-5c8d4596ab24", :type=>:customer}}}}}
(byebug) ActivitySerializer.new([activity])
#<ActivitySerializer:0x00005564382f2078 @fieldsets={}, @params={}, @resource=[#<Activity id: "ab530f49-47df-4208-af21-0238031bb66e", user_id: "f953401d-37d4-4780-b471-5c8d4596ab24", resource_type: "User", resource_id: "f953401d-37d4-4780-b471-5c8d4596ab24", action: "POST", client: "browser", os: "Windows with SQL Server Web (Amazon VPC)", remote_address: #<IPAddr: IPv4:144.91.96.43/255.255.255.255>, misc: {}, created_at: "2020-12-06 18:24:34">]>
(byebug) ActivitySerializer.new([activity]).to_hash
{:data=>[{:id=>"ab530f49-47df-4208-af21-0238031bb66e", :type=>:activity, :attributes=>{:action=>"POST", :client=>"browser", :os=>"Windows with SQL Server Web (Amazon VPC)", :misc=>{}, :created_at=>Sun, 06 Dec 2020 18:24:34 UTC +00:00, :remote_address=>"144.91.96.43"}, :relationships=>{:user=>{:data=>{:id=>"f953401d-37d4-4780-b471-5c8d4596ab24", :type=>:customer}}, :resource=>{:data=>{:id=>"f953401d-37d4-4780-b471-5c8d4596ab24", :type=>:customer}}}}]}
(byebug) ActivitySerializer.ancestors
[ActivitySerializer, ActionView::Helpers, ActionView::Helpers::RenderingHelper, ActionView::Helpers::NumberHelper, ActionView::Helpers::JavaScriptHelper, ActionView::Helpers::FormOptionsHelper, ActionView::Helpers::TranslationHelper, ActionView::Helpers::FormHelper, ActionView::RecordIdentifier, ActionView::ModelNaming, ActionView::Helpers::FormTagHelper, ActionView::Helpers::TextHelper, ActionView::Helpers::DebugHelper, ActionView::Helpers::SanitizeHelper, ActionView::Helpers::CacheHelper, ActionView::Helpers::AtomFeedHelper, ActionView::Helpers::UrlHelper, ActionView::Helpers::AssetTagHelper, ActionView::Helpers::AssetUrlHelper, ActionView::Helpers::ActiveModelHelper, ActiveSupport::Benchmarkable, ActionView::Helpers::TagHelper, ActionView::Helpers::OutputSafetyHelper, ActionView::Helpers::DateHelper, ActionView::Helpers::CsrfHelper, ActionView::Helpers::CspHelper, ActionView::Helpers::ControllerHelper, ActionView::Helpers::CaptureHelper, BaseSerializer, FastJsonapi::ObjectSerializer, FastJsonapi::SerializationCore, JSONAPI::Serializer, ActiveSupport::Dependencies::ZeitwerkIntegration::RequireDependency, ActiveSupport::ToJsonWithActiveSupportEncoder, Object, PP::ObjectMixin, ActiveSupport::Tryable, ActiveSupport::Dependencies::Loadable, JSON::Ext::Generator::GeneratorMethods::Object, Kernel, BasicObject]
```

**Comment by 0x962** [comment]
This error does not seem to occur when invoking the serialiser in a rails console or via byebug. I've only been able to reproduce this by putting it in a controller. 

```
 module LiveApi
   module V1
     class SkusController < ApplicationController

       def index
         skus = Sku.all
         render json: SkuSerializer.new(skus, {include: ['item']})
       end

     end

   end
 end
```

I've uploaded a trace here: https://drive.google.com/file/d/13_ZlQYelBRzkvuDEtD3H9bmFNkSqFwSC/view?usp=sharing

**Comment by 0x962** [comment]
Unfortunately, I don't have much else either. I couldn't find any meaningful references in the trace either, except that adding `ActionView::Helpers` causes the problem, removing it fixes things. 

I will close this issue now.

**Comment by stas** [comment]
Please review your serializer methods/attributes, if there's any conflicting method/call, that might be the issue. Try to comment out any of the serializer attribute/relationship definitions and see when it breaks eventually. Sorry for not being able to help much.

**Comment by aximuseng** [comment]
I ran into the exact same issue (related to issue #157) once I added the `includes`. It turns out that the issue was in reality Devise - my authentication had expired.  I had not excluded devise in my base API controller:

`skip_before_action :authenticate_user!`

Once I did this the stack depth errors went away.  I also realized that my key based API would have been use useless as it would have only worked if a valid user was logged in on the same system 🤦‍♂️

My be unrelated but perhaps it will help someone.
